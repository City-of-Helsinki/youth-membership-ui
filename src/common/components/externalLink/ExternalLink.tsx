import React from 'react';
import { useTranslation } from 'react-i18next';
import { IconLinkExternal } from 'hds-react';

import styles from './externalLink.module.css';

type Props = React.PropsWithChildren<{
  href: string;
  className?: string;
}>;

const ExternalLink = ({ href, children, className, ...rest }: Props) => {
  const { t } = useTranslation();

  return (
    <a
      href={href}
      target="_blank"
      rel="noopener noreferrer"
      {...rest}
      className={[styles.externalLink, className].join(' ')}
    >
      {children} <IconLinkExternal aria-label={t('externalLink.description')} />
    </a>
  );
};

export default ExternalLink;
