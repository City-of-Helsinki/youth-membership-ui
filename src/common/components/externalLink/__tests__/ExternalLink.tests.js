import React from 'react';

import { render } from '../../../test/testing-library';
import ExternalLink from '../ExternalLink';

const defaultProps = {
  href: 'https://hel.fi',
  children: 'Helsingin kaupunki',
};
const getWrapper = props =>
  render(<ExternalLink {...defaultProps} {...props} />);

describe('<ExternalLink />', () => {
  it('renders correctly', () => {
    expect(getWrapper().container).toMatchSnapshot();
  });

  it('should warn that it opens in a new window', () => {
    const { getByLabelText } = getWrapper();

    expect(getByLabelText('Aukeaa uuteen ikkunaan')).toBeInTheDocument();
  });
});
