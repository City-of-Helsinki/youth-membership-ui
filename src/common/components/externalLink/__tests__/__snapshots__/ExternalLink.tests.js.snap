// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExternalLink /> renders correctly 1`] = `
<div>
  <a
    class="externalLink "
    href="https://hel.fi"
    rel="noopener noreferrer"
    target="_blank"
  >
    Helsingin kaupunki
     
    <svg
      aria-label="Aukeaa uuteen ikkunaan"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M10 3v2H5v14h14v-5h2v7H3V3h7zm11 0v8h-2V6.413l-7 7.001L10.586 12l6.999-7H13V3h8z"
          fill="currentColor"
        />
      </g>
    </svg>
  </a>
</div>
`;
